php/class-photonfill.php

```diff
@@ -368,6 +368,61 @@ public function create_image_object( $attachment_id, $current_size, $args = arra
 		}
 
 		/**
+		 * Create the necessary data structure for an external url image.
+		 * @param string $img_url
+		 * @param array $sizes
+		 * @param string $arg Optional args. defined args are currently
+		 * @return array
+		 */
+		public function create_url_image_object( $img_url, $current_size, $args = array() ) {
+			if ( ! empty( $img_url ) ) {
+				$sizes = $this->image_sizes;
+				$image_sizes = array();
+
+				if ( ! is_array( $current_size ) && ! empty( $sizes[ $current_size ] ) ) {
+					foreach ( $sizes[ $current_size ] as $breakpoint => $img_size ) {
+						$default = ( ! empty( $img_size['default'] ) ) ? true : false;
+						$current_w = empty( $img_size['width'] ) ? 0 : $img_size['width'];
+						$current_h = empty( $img_size['height'] ) ? 0 : $img_size['height'];
+						$transform_args = array(
+							'callback' => ( isset( $img_size['callback'] ) ) ? $img_size['callback'] : null,
+							'crop' => ( isset( $img_size['crop'] ) ) ? $img_size['crop'] : true,
+							'breakpoint' => $breakpoint,
+							'image_size' => $current_size,
+							'width' => $current_w,
+							'height' => $current_h,
+							'quality' => ( isset( $img_size['quality'] ) ) ? $img_size['quality'] : null,
+						);
+						$this->transform->setup( $transform_args );
+						$img_src = $this->get_url_img_src( $img_url, array( $current_w, $current_h ), $default );
+						$image_sizes[ $breakpoint ] = array( 'size' => $this->breakpoints[ $breakpoint ], 'src' => $img_src );
+					}
+				} elseif ( is_array( $current_size ) ) {
```

Comment:
It may just be me, but I think this code could benefit from comments that explain the WHY behind it, to help the reader understand the purpose behind each condition.